core/keyboard_nav/key_map.js

```diff
@@ -0,0 +1,250 @@
+/**
+ * @license
+ * Visual Blocks Editor
+ *
+ * Copyright 2019 Google Inc.
+ * https://developers.google.com/blockly/
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * @fileoverview The namespace used to keep track of keyboard actions and the
+ * key codes used to execute those actions.
+ * This is used primarily for keyboard navigation.
+ */
+'use strict';
+
+goog.provide('Blockly.user.keyMap');
+
+/**
+ * Holds the serialized key to key action mapping.
+ * @type {Object<string, Blockly.KeyAction>}
+ */
+Blockly.user.keyMap.map_ = {};
+
+/**
+ * Holds the list of actions that can be executed using the keyboard.
+ * @type {Array<Blockly.KeyAction>}
+ */
+Blockly.user.keyMap.actions_ = [];
```

Comment:
You aren't reading this anywhere. I don't think you need it.

Reply:
done